metasyn/config.py

```diff
@@ -0,0 +1,175 @@
+                             f" Unknown keys detected: '{list(general)}'")
+        return cls(var_list, dist_providers, privacy, n_rows=n_rows)
+
+    def to_dict(self) -> dict:
+        """Convert the configuration to a dictionary.
+
+        Returns
+        -------
+        config_dict:
+            Configuration in dictionary form.
+        """
+        return {
+            "general": {
+                "privacy": self.privacy,
+                "dist_providers": self.dist_providers,
+            },
+            "var": self.var_configs
+        }
+
+    def get(self, name: str) -> VarConfigAccess:
+        """Create a VarConfigAccess object pointing to a var with that name.
+
+        If the variable does not exist, then a new variable config is created that
+        has the default values.
+
+        Parameters
+        ----------
+        name:
+            Name of the variable configuration to retrieve.
+
+        Returns
+        -------
+        var_cfg:
+            A variable config access object.
+        """
+        for var_cfg in self.var_configs:
+            if var_cfg.name == name:
+                return VarConfigAccess(var_cfg, self)
+        return VarConfigAccess(VarConfig(name=name), self)
+
+    def iter_var(self, exclude: Optional[list[str]] = None) -> Iterable[VarConfigAccess]:
+        """Iterate over all variables in the configuration.
+
+        Parameters
+        ----------
+        exclude:
+            Exclude variables with names in that list.
+
+        Returns
+        -------
+        var_cfg:
+            VarConfigAccess class for each of the available variable configurations.
+        """
+        exclude = exclude if exclude is not None else []
+        for var_spec in self.var_configs:
+            if var_spec.name not in exclude:
+                yield VarConfigAccess(var_spec, self)
+
+
+class VarConfigAccess():  # pylint: disable=too-few-public-methods
+    """Access for variable configuration object.
+
+    They take into account what the defaults are from the MetaConfig object.
+    Otherwise they pass through all the attributes as normal and thus behave
+    exactly as a variable config object themselves.
+
+    Parameters
+    ----------
+    var_config
+        The variable configuration to access.
+    meta_config
+        The meta configuration instance to get default values from.
+    """
+
+    def __init__(self, var_config: VarConfig, meta_config: MetaConfig):
+        self.var_config = var_config
+        self.meta_config = meta_config
+
+    def __getattribute__(self, attr):
+        if attr == "privacy":
+            if self.var_config.privacy is None:
+                return self.meta_config.privacy
+            return self.var_config.privacy
+        if attr not in ("var_config", "meta_config") and hasattr(self.var_config, attr):
+            return getattr(self.var_config, attr)
+        return super().__getattribute__(attr)
```

Comment:
I think this is kind of strange. So this is a class around the class just to get the attributes of the class?? Can't we put this in MetaConfig or VarConfig directly?